x-pack/platform/plugins/shared/streams/server/routes/content/route.ts

```diff
@@ -62,48 +55,46 @@ const exportContentRoute = createServerRoute({
     },
   },
   async handler({ params, request, response, getScopedClients, context }) {
-    const { assetClient, soClient, streamsClient } = await getScopedClients({ request });
+    const { assetClient, streamsClient } = await getScopedClients({ request });
 
-    await streamsClient.ensureStream(params.path.name);
-
-    if (!isIncludeAll(params.body.include) && params.body.include.objects.dashboards.length === 0) {
-      throw new StatusError(`Content pack must include at least one object`, 400);
+    const root = await streamsClient.getStream(params.path.name);
+    if (!Streams.WiredStream.Definition.is(root)) {
+      throw new StatusError('Only wired streams can be exported', 400);
     }
 
-    function isDashboard(asset: Asset): asset is DashboardAsset {
-      return asset[ASSET_TYPE] === 'dashboard';
-    }
-
-    const dashboards = (await assetClient.getAssets(params.path.name))
-      .filter(isDashboard)
-      .filter(
-        (dashboard) =>
-          isIncludeAll(params.body.include) ||
-          params.body.include.objects.dashboards.includes(dashboard['asset.id'])
-      );
-    if (dashboards.length === 0) {
-      throw new StatusError('No included objects were found', 400);
-    }
+    const [ancestors, descendants] = await Promise.all([
+      streamsClient.getAncestors(params.path.name),
+      streamsClient.getDescendants(params.path.name),
+    ]);
 
-    const exporter = (await context.core).savedObjects.getExporter(soClient);
-    const exportStream = await exporter.exportByObjects({
-      request,
-      objects: dashboards.map((dashboard) => ({ id: dashboard[ASSET_ID], type: 'dashboard' })),
-      includeReferencesDeep: true,
+    const exportedDescendants = isIncludeAll(params.body.include)
+      ? descendants
+      : resolveAncestors(params.body.include.objects.streams).map((name) => {
+          const descendant = descendants.find((d) => d.name === withRootPrefix(root.name, name));
+          if (!descendant) {
+            throw new StatusError(
+              `Could not find [${name}] as a descendant of [${root.name}]`,
+              400
+            );
+          }
+          return descendant;
+        });
+
+    const inheritedFields = getInheritedFieldsFromAncestors(ancestors);
```

Comment:
It seems like the inherited fields are not turned into regular field definitions - they still have the `from` in there which they shouldn't right? Am I missing something here?